tests/parser-test-files/define-class.txt

```diff
@@ -13,3 +336,154 @@ error: Malformed class definition
+
+error: Malformed functional dependency
+  --> test:3:34
+   |
+ 3 |  (define-class (C :a :b (:a -> :b) :c))
+   |                                    ^^ expected a list
+
+================================================================================
+115 Malformed functional dependency: dotted list
+================================================================================
+
+(package test-package)
+
+(define-class (C :a :b (:a :b . :c)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:23
+   |
+ 3 |  (define-class (C :a :b (:a :b . :c)))
+   |                         ^^^^^^^^^^^^ unexpected dotted list
+
+================================================================================
+116 Malformed functional dependency: missing type variables
+================================================================================
+
+(package test-class)
+
+(define-class (C :a :b (:a :b ->)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:33
+   |
+ 3 |  (define-class (C :a :b (:a :b ->)))
+   |                                  ^ expected one or more type variables
+
+================================================================================
+117 Malformed functional dependency: missing type variables (2)
+================================================================================
+
+(package test-package)
+
+(define-class (C :a :b (-> :b :c)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:23
+   |
+ 3 |  (define-class (C :a :b (-> :b :c)))
+   |                         ^^^^^^^^^^ expected one or more type variables
+
+================================================================================
+118 Invalid attribute for define-class: illegal attribute
+================================================================================
+
+(package test-package)
+
+(repr :enum)
+(define-class (C :a))
+
+--------------------------------------------------------------------------------
+
+error: Invalid attribute for define-class
+  --> test:3:0
+   |
+ 3 |  (repr :enum)
+   |  ^^^^^^^^^^^^ define-class cannot have attributes
+ 4 |  (define-class (C :a))
+   |                ------ while parsing define-class
+
+================================================================================
+119 Malformed class definition
+================================================================================
+
+(package test-package)
+
+(define-class (Eq :a => (C :a) :b))
+
+--------------------------------------------------------------------------------
+
+error: Malformed class definition
+  --> test:3:31
+   |
+ 3 |  (define-class (Eq :a => (C :a) :b))
+   |                                 ^^ unexpected form
```

Comment:
Would be better to have a "trailing" error, though might be awkward to implement